### PR TITLE
0.5.7

### PR DESCRIPTION
**Pull request by ca333**
### Release Notes 

**Features:** 

* :pirate_flag:	https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1775
* :green_book: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1921
* :dna: Updated API core to [beta-2.1.8741](https://github.com/KomodoPlatform/atomicDEX-API/releases/tag/beta-2.1.8741) 
* :de: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1939
* :es: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2010
* :moneybag:  Add the following Smartchain coins: `ARRR`, `ZOMBIE`, `ILNF`
* :moneybag:  Add the following utxo coins: `CHTA` `LNC` `AVN` `NENG`
* :moneybag:  Add the following test coins: `ZOMBIE`
* :moneybag:  Add the following BEP-20 tokens: `ILNF` `JST` `LOOM` `LRC` `BTCZ` `WOO`
* :moneybag:  Add the following ERC-20 tokens: `LOOM` `TAMA` `GLM` `LDO` `WOO`
* :moneybag:  Add the following PLG-20 tokens: `ILNF` `ILNSW` `ANKR` `APE` `ATOM` `BRZ` `GLM` `KNC` `LDO` `SOL` `WOO` `YFI` 
* :moneybag:  Add the following FTM-20 tokens: `APE` `WOO` 
* :moneybag:  Add the following AVX-20 tokens: `BRZ` `GMX` `KNC` `WOO` 
* :moneybag:  Add the following KRC-20 tokens: `SHIB` 
* :moneybag:  Add the following SLP tokens: `USDT`  `TTT` `ASLP` `SPICE` `FLEXUSD` `SOUR` `PSF` `MAZE` `USDF`
* :moneybag:  Delisted the following coins: `ABY-OLD` `ZET-OLD` `JSTR` `ETHK-OPT20` `ETHR-ARB20` `ETHR`

**Enhancements:**

* :mage:  Streamline coin listing/delisting by sourcing directly from [coins repository](https://github.com/KomodoPlatform/coins)
* :warning: Improved validation & error messages 
* :lab_coat: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1892 https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1908
* :warning: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1934
* :hourglass_flowing_sand: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1945
* :tropical_fish: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1944
* :stop_sign: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1958
* :telescope: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1983
* :children_crossing: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1937
* :sunglasses: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2042
* 🧹  https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2043
* 🖥️  https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1961 
 
**Fixes:**
* :chart_with_upwards_trend: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1886
* :fountain_pen: Updated OSX code signing
* :biohazard: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1900
* :package:	https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2061 
* :shoe: Fixed misc visual bugs 
* :no_entry: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1954
* :plate_with_cutlery: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/1960
* :butterfly: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2038 
* :lotus_position_woman: https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2008

### CODE REVIEW

**Review by DeckerSU — DISMISSED** [review]
LGTM.

**p.s.** PR assume version bump to `0.5.7`, but `get_version()`, `get_raw_version()`, etc. from `version.hpp` still returning old version `0.5.6`.
**p.p.s.** PR verified from a "security side", logic and implementation errors didn't tested for obvious reasons (time, deep-in, etc.)

File: atomic_defi_design/Dex/Components/ZcashParamsModal.qml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                text: qsTr("Close")
+                onClicked: close()
+            },
+            Item { Layout.fillWidth: true }
+        ]
+
+        Connections
+        {
+            target: Dex.API.app.zcash_params
+            function onCombinedDownloadStatusChanged()
```
Comment (SylEze) [inline atomic_defi_design/Dex/Components/ZcashParamsModal.qml:92]: This implementation is slow and error prone, it should be done in C++ where there is strong type safety: put at least a todo for next releases

File: atomic_defi_design/Dex/Exchange/ProView/PlaceOrderForm/Main.qml [hunk 2]
```diff
@@ -17,19 +17,95 @@ Widget
...
+        Layout.preferredWidth: parent.width
+        visible: protocolIcon != ""
+
+        ColumnLayout
+        {
+            spacing: 2
+            anchors.fill: parent
+            anchors.centerIn: parent
+
+            DexLabel
```
Comment (SylEze) [inline atomic_defi_design/Dex/Exchange/ProView/PlaceOrderForm/Main.qml:63]: Use `Dex.Text`.

File: src/app/app.cpp [hunk 3]
```diff
@@ -76,25 +76,30 @@ namespace atomic_dex
...
             auto coin_info = mm2.get_coin_info(coin.toStdString());
-            if (coin_info.has_parent_fees_ticker && coin_info.ticker != coin_info.fees_ticker)
+            
+            if (coin_info.has_parent_fees_ticker &&
+                coin_info.ticker != coin_info.fees_ticker &&
+                !coins.contains(QString::fromStdString(coin_info.fees_ticker)))
             {
                 auto coin_parent_info = mm2.get_coin_info(coin_info.fees_ticker);
-                if (!coin_parent_info.currently_enabled && !coin_parent_info.active && extra_coins.insert(coin_parent_info.ticker).second)
+                if (coin_parent_info.ticker != "")
```
Comment (SylEze) [inline src/app/app.cpp:88]: weird. parent's ticker should not empty if it has been found.

**Review by SylEze — CHANGES_REQUESTED** [review]

File: atomic_defi_design/Dex/Components/CoinMenu.qml [hunk 4]
```diff
@@ -28,7 +28,15 @@ Menu {
...
         onTriggered: API.app.disable_coins([ticker])
-        enabled: General.canDisable(ticker)
+        enabled:
+        {
+            let progress = General.zhtlcActivationProgress(current_ticker_infos.activation_status, ticker)
+            if (General.isZhtlc(ticker))
+            {
+                if (progress != 100) return false
+            }
+            return General.canDisable(ticker, progress)
```
Comment (SylEze) [inline atomic_defi_design/Dex/Components/CoinMenu.qml:38]: canDisable does not accept two parameters
And even if it's the case it should not need a ZHTLC `progress` param

File: atomic_defi_design/Dex/Components/CoinMenu.qml [hunk 5]
```diff
@@ -28,7 +28,15 @@ Menu {
         id: disable_action
         text: qsTr("Disable %1", "TICKER").arg(ticker)
         onTriggered: API.app.disable_coins([ticker])
-        enabled: General.canDisable(ticker)
+        enabled:
+        {
+            let progress = General.zhtlcActivationProgress(current_ticker_infos.activation_status, ticker)
```
Comment (SylEze) [inline atomic_defi_design/Dex/Components/CoinMenu.qml:33]: ZHTLC should not be called when the current ticker is not a ZHTLC asset...
Put this statement inside the if below ?

Comment (SylEze) [reply atomic_defi_design/Dex/Components/CoinMenu.qml:33]: Maybe we can put the zhtlc logic inside the canDisable function directly

File: atomic_defi_design/Dex/Exchange/ProView/PlaceOrderForm/OrderFormSubfield.qml [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+import "../../../Constants" as Dex
+import "../../../Components"
+import App 1.0
+import Dex.Themes 1.0 as Dex
+
+RowLayout
+{
+    id: control
+    property alias  fiat_value: _fiat_label.text_value
```
Comment (SylEze) [inline atomic_defi_design/Dex/Exchange/ProView/PlaceOrderForm/OrderFormSubfield.qml:14]: Coding style is wrong: use camelCase

**Review by SylEze — CHANGES_REQUESTED** [review]

**Comment by smk762** [comment]
image
SLP token mainnet confirmed

**Review by SylEze — APPROVED** [review]

File: CMakeLists.txt [hunk 7]
```diff
@@ -61,13 +61,13 @@ endif ()
 ##! We fetch our dependencies
 if (APPLE)
     FetchContent_Declare(mm2
-            URL https://github.com/KomodoPlatform/atomicDEX-API/releases/download/beta-2.1.7404/mm2-0f6c72615-Darwin-Release.zip)
+            URL http://195.201.0.6/mm2.1/mm2-1e1338d2d-Darwin-Release.zip)
```
Comment (tonymorony) [inline CMakeLists.txt:64]: we need API release from sec env before merge

**Review by smk762 — APPROVED** [review]
Thanks to all involved with this mammoth release! :revolving_hearts:

**Review by SirSevenG — APPROVED** [review]

**Review by DeckerSU — APPROVED** [review]

**Pull request merged**